src/common/regions/region_hooks.ts

```diff
@@ -0,0 +1,50 @@
+import { useContext } from 'react';
+import { useParams } from 'react-router';
+import regions, { Region, RegionContext } from 'common/regions';
+import { assert } from 'common/utils';
+
+export const useLocationPageRegion = () => {
+  const region = useContext(RegionContext);
+  assert(
+    region,
+    '`useLocationPageRegion` can only be called from components inside LocationPage',
+  );
+  return region;
+};
+
+export const useRegionFromLegacyIds = (
+  stateId: string,
+  countyId?: string,
+  countyFipsId?: string,
+) => {
+  if (countyFipsId) {
+    return regions.findByFipsCode(countyFipsId);
+  }
+  const state = regions.findStateByUrlParams(stateId);
+  const county = countyId
+    ? regions.findCountyByUrlParams(stateId, countyId)
+    : null;
+  return county || state;
+};
```

Comment:
I would recommend not implementing these hooks as part of the refactor, they might be convenient but I don't think is going to help us with MSAs